security/src/main/java/com/devonfw/tools/security/BuildSecurityJsonFiles.java

```diff
@@ -0,0 +1,356 @@
+      NodeAuditAnalyzer.class, YarnAuditAnalyzer.class, PnpmAuditAnalyzer.class, RetireJsAnalyzer.class,
+      FalsePositiveAnalyzer.class);
+
+  private static BigDecimal minV2Severity;
+
+  private static BigDecimal minV3Severity;
+
+  private static final Set<String> actuallyIgnoredCves = new HashSet<>();
+
+  private static IdeContext context;
+
+  /**
+   * @param args Set {@code minV2Severity} with {@code args[0]} and {@code minV3Severity} with {@code args[1]}.
+   *        Vulnerabilities with severity lower than these values will not be added to {@link UrlSecurityJsonFile}.
+   */
+  public static void main(String[] args) {
+
+    if (args.length != 2) {
+      throw new RuntimeException("Please provide 2 numbers: minV2Severity and minV3Severity");
+    }
+    try {
+      minV2Severity = new BigDecimal(String.format(args[0]));
+      minV3Severity = new BigDecimal(String.format(args[1]));
+    } catch (NumberFormatException e) {
+      throw new RuntimeException("These two args could not be parsed as BigDecimal");
+    }
+    run();
+
+  }
+
+  private static void run() {
+
+    initCvesToIgnore();
+    context = new IdeContextConsole(IdeLogLevel.INFO, null, false);
+    UpdateManager updateManager = new UpdateManager(context.getUrlsPath(), null);
+    Dependency[] dependencies = getDependenciesWithVulnerabilities(updateManager);
+    Set<Pair<String, String>> foundToolsAndEditions = new HashSet<>();
+    for (Dependency dependency : dependencies) {
+      String filePath = dependency.getFilePath();
+      Path parent = Paths.get(filePath).getParent();
+      String tool = parent.getParent().getParent().getFileName().toString();
+      String edition = parent.getParent().getFileName().toString();
+      AbstractUrlUpdater urlUpdater = updateManager.getUrlUpdater(tool);
+      UrlSecurityJsonFile securityFile = context.getUrls().getEdition(tool, edition).getSecurityJsonFile();
+      boolean newlyAdded = foundToolsAndEditions.add(new Pair<>(tool, edition));
+      if (newlyAdded) { // to assure that the file is cleared only once per tool and edition
+        securityFile.clearSecurityWarnings();
+      }
+
+      List<String> sortedVersions = context.getUrls().getSortedVersions(tool, edition).stream()
+          .map(VersionIdentifier::toString).toList();
+      List<String> sortedCpeVersions = sortedVersions.stream().map(urlUpdater::mapUrlVersionToCpeVersion)
+          .collect(Collectors.toList());
+      Map<String, String> cpeToUrlVersion = MapUtil.createMapfromLists(sortedCpeVersions, sortedVersions);
+
+      Set<Vulnerability> vulnerabilities = dependency.getVulnerabilities(true);
+      for (Vulnerability vulnerability : vulnerabilities) {
+        addVulnerabilityToSecurityFile(vulnerability, securityFile, urlUpdater, cpeToUrlVersion);
+      }
+      securityFile.save();
+    }
+    actuallyIgnoredCves.forEach(cve -> context.info("Ignored CVE " + cve + " because it is listed in CVES_TO_IGNORE."));
```

Comment:
The regular user probably doesn't have to interact with this class. Because we run it and push the security files to the `ide-urls` repo. I would leave it in here. We can reconsider this if we have got many CVEs to ignore. Or we create an issue that implements different log levels for this project.